lib/node.js

```diff
@@ -1208,4 +1245,70 @@ class Node {
   }
 }
 
+/**
+ * Internal class that creates Timer instances in a common private rcl context
+ * for use with Rate. The private rcl context ensures that Rate timers do not
+ * deadlock waiting for spinOnce/spin on the main rcl context.
+ */
+class RateTimerServer {
```

Comment:
Since this only relates to Rates, can this class go in the `rate.js` file instead? The `node.js` file is already very large.

Other than that it LGTM.

Reply:
Agree node.js is getting large and I originally I implemented RateTimeServer in rate.js for that reason. But at the last min I moved it node.js to keep it a private and void exporting it from Rate. I'm fine with moving RateTimeServer to rate and exporting it.

Reply:
The latest commit includes RateTimeServer relocated from node.js to rate.js